src/main/java/com/sem/btrouble/model/SlowPowerUp.java

```diff
@@ -9,40 +9,69 @@
 
 import com.sem.btrouble.event.BubbleEvent;
 
+/**
+ * Represents power up for slow bubbles.
+ * @author Martin
+ *
+ */
+@SuppressWarnings("serial")
 public class SlowPowerUp extends PowerUp implements Observer{
     
     private boolean on;
 	private Image playerIdle;
     
+	/**
+	 * Construct power up from shop.
+	 */
     public SlowPowerUp() {
     	super();
     	on = false;
     	activate();
     }
     
+    /**
+     * Construct power up received in the game.
+     * @param xpos x position
+     * @param ypos y position
+     */
     public SlowPowerUp(float xpos, float ypos) {
         super(xpos, ypos);
         on = false;
     }
     
+    /**
+     * Update the speed of the bubbles.
+     * @param observable object to observe
+     * @param arg the event
+     */
     public void update(Observable observable, Object arg) {
       if (arg instanceof BubbleEvent) {
           BubbleEvent event = (BubbleEvent) arg;
-          if (event.getId() == BubbleEvent.COLLISION_ROPE && on == true) {
+          if (event.getId() == BubbleEvent.COLLISION_ROPE && on) {
               slowBubbles(.3f);
```

Comment:
The same code is run on a Observer update, and in the activate() method. 
Why is slowBubbles also in the Observer update?